src/app/containers/App/App.jsx

```diff
@@ -55,28 +56,15 @@ export const App = ({ routes, location, initialData, bbcOrigin, history }) => {
         error: null,
       });
 
-      const fetchData = async () => {
-        try {
-          const {
-            pageData: _pageData,
-            status: _status,
-          } = await route.getInitialData(match.params);
-
-          setState(prevState => ({
-            ...prevState,
-            pageData: _pageData,
-            status: _status,
-            loading: false,
-          }));
-        } catch (error) {
-          setState(prevState => ({
-            ...prevState,
-            error,
-            loading: false,
-          }));
-        }
-      };
-      fetchData();
+      route.getInitialData(match.params).then(data =>
+        setState(prevState => ({
+          ...prevState,
+          loading: false,
+          pageData: path(['pageData'], data),
+          status: path(['status'], data),
+          error: path(['error'], data),
```

Comment:
We need to handle `error` on line 28 too. As it stands, errors will be handled on onward navigation differently to the initial server render.

Reply:
@dr3 fixed by destructuring `error` from `initialData` and using it in the `initialState`.